### DISCUSSION TITLE
Does SOCI index support GHCR?

### DISCUSSION THREAD

**Discussion by gyanesh-mishra** [awslabs/soci-snapshotter#979] [Q&A]
I have an image that i'm building the SOCI indexes for on both ECR and GHCR.
However, when I pull the image from ECR, I get lazy loading but not from GHCR.

What could be the cause here?
I'm using SOCI 0.4.0 and this is the GHCR repo.
https://github.com/predibase/lorax/pkgs/container/lorax

**Comment by ollypom** [comment]
According to the [Registry Docs](https://github.com/awslabs/soci-snapshotter/blob/main/docs/registry.md), GHCR has been tested and works with SOCI.

Out of curiosity how are you attempting to run the container image? Are you using AWS Fargate?

**Comment by gyanesh-mishra** [comment]
Im using EKS for this and have been using it successfully with ECR for a few weeks now.
Do I need to use the legacy registry option? 🤔

**Reply by Kern--** [reply]
No. The `--legacy-registry` option was removed before v0.1.0. It was only necessary really early on when the OCI Image Spec was considering adding a new "Artifact Manifest" type. Eventually, they decided to store artifacts (like SOCI indexes) using the existing Image Manifest, so you don't need a flag based on registry support. SOCI will either automatically detect the right way to store the index in the registry or the registry isn't supported by SOCI.

**Comment by Kern--** [comment]
You might be able to find some more information if you inspect the SOCI snapshotter's logs.

Credentials is one place that I could imagine a difference based on registry.

**Reply by gyanesh-mishra** [reply]
Oh interesting! I'll take a look.
The GHCR repo that I am pulling from is a public repo so I didn't think creds would be an issue.

Can the same snapshotter pull from a Private ECR and a Public GHCR and lazy load both?

Since this is an EKS cluster there is no gurantee which node will get the pod/image assigned to it.

**Reply by Kern--** [reply]
Yes. As long as it can find credentials it can pull from both (e.g. your docker config has credential providers/credentials configured for both).

IIRC, GHCR doesn't allow anonymous pulls. Any authenticated user can pull from a public repository, but you have to have a valid token.
